content/src/service/access/AccessCheckerForWearables.ts

```diff
@@ -15,62 +15,114 @@ export class AccessCheckerForWearables {
-        const hasAccess = await this.checkCollectionAccess(
-          blocksSubgraphUrl,
-          collectionsSubgraphUrl,
-          collection,
-          itemId,
-          accessParams
-        )
-        if (!hasAccess) {
-          errors.push(`The provided Eth Address does not have access to the following wearable: (${pointer})`)
-        }
-      } else {
-        errors.push(
-          `Wearable pointers should be a urn, for example (urn:decentraland:{protocol}:collections-v2:{contract(0x[a-fA-F0-9]+)}:{name}). Invalid pointer: (${pointer})`
-        )
-      }
+      const { contractAddress: collection, id: itemId } = parsed
+      const collectionsSubgraphUrl = isL1Network ? this.collectionsL1SubgraphUrl : this.collectionsL2SubgraphUrl
+      const blocksSubgraphUrl = isL1Network ? this.blocksL1SubgraphUrl : this.blocksL2SubgraphUrl
+      // Check that the address has access
+      const hasAccess = await this.checkCollectionAccess(
+        blocksSubgraphUrl,
+        collectionsSubgraphUrl,
+        collection,
+        itemId,
+        accessParams
+      )
+      if (!hasAccess) return [`The provided Eth Address does not have access to the following wearable: (${pointer})`]
     }
-    return errors
+    return []
   }
 
-  private async parseUrnNoFail(urn: string): Promise<BlockchainCollectionV2Asset | null> {
+  private async parseUrnNoFail(
+    urn: string
+  ): Promise<BlockchainCollectionV2Asset | BlockchainCollectionThirdParty | null> {
     try {
       const parsed = await parseUrn(urn)
       if (parsed?.type === 'blockchain-collection-v2-asset') {
         return parsed as BlockchainCollectionV2Asset
+      } else if (parsed?.type === 'blockchain-collection-third-party') {
+        return parsed as BlockchainCollectionThirdParty
       }
     } catch {}
     return null
   }
 
+  private async checkItemExistence(
+    urn: string,
+    collectionsSubgraphUrl: string,
+    blocksSubgraphUrl: string,
+    { timestamp, content, metadata }: Omit<WearablesAccessParams, 'pointers'>
+  ): Promise<boolean> {
+    try {
+      const { blockNumberAtDeployment, blockNumberFiveMinBeforeDeployment } = await this.findBlocksForTimestamp(
+        blocksSubgraphUrl,
+        timestamp
+      )
+
+      const existsOnBlock = async (blockNumber: number | undefined) =>
+        !!blockNumber && this.existsItem(collectionsSubgraphUrl, urn, blockNumber, content, metadata)
+
+      return (await existsOnBlock(blockNumberAtDeployment)) || (await existsOnBlock(blockNumberFiveMinBeforeDeployment))
+    } catch (error) {
+      this.LOGGER.error(`Error checking item existence (${urn}).`, error)
+      return false
+    }
+  }
+
+  private async existsItem(
+    subgraphUrl: string,
+    urn: string,
+    block: number,
+    content?: Map<string, string>,
+    metadata?: any
+  ): Promise<boolean> {
+    const query = `
+    query ($urn: String!, $block: Int!) {
+      items(where:{ urn: $urn}, block: { number: $block }) {
```

Comment:
should the `contentHash` be in the where clause?

Reply:
Also, are we considering a 5 minutes window for checking permissions as we are doing on other wearables collections?